docs/guides/user-containers/readme.md

```diff
@@ -1,96 +1,181 @@
+Workflow](#creating-a-container-workflow).
 
-## Container Ports
+Storages may deemed be `optional` in a profile. If a storage is not option, the storage name
+must be set the name of an NNF filesystem name in the container workflow.
 
-NNF Container Profiles allow for ports to be reserved for a container workflow. `numPorts` can be used to specify the number of ports needed for a container workflow:
+For global lustre, there is an additional field for `pvcMode`, which must match the mode that is
+configured in the `LustreFilesystem` resource that represents the global lustre filesystem. This
+defaults to `ReadWriteMany`.
+
+Example:
 
 ```yaml
-kind: NnfContainerProfile
-metadata:
-  name: sample-nnfcontainerprofile
-  namespace: nnf-system
-data:
-  # Request the number of ports to open on the targeted rabbits. These ports are accessible outside
-  # of the k8s cluster.  The requested ports are made available as environment variables inside the
-  # container and in the DWS workflow (NNF_CONTAINER_PORTS).
-  numPorts: 1
+  storages:
+  - name: DW_JOB_foo_local_storage
+    optional: false
+  - name: DW_PERSISTENT_foo_persistent_storage
+    optional: true
+  - name: DW_GLOBAL_foo_global_lustre
+    optional: true
+    pvcMode: ReadWriteMany
 ```
 
-The allocated port numbers are made available via the [`NNF_CONTAINER_PORTS`](#nnf_container_ports) environment variable.
+#### Container Spec
+
+As mentioned earlier, container workflows can be categorized into two types: MPI and Non-MPI. It's
+essential to choose and define only one of these types within the container profile. Regardless of
+the type chosen, the data structure that implements the specification is equipped with two
+"standard" resources that are distinct from NNF custom resources.
+
+For Non-MPI containers, the specification utilizes the `spec` resource. This is the standard
+Kubernetes
+[`PodSpec`](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec)
+that outlines the desired configuration for the pod.
+
+For MPI containers, `mpiSpec` is used. This custom resource, available through
+[`MPIJobSpec`]
+from `mpi-operator`, serves as a facilitator for executing MPI applications across worker
+containers. This resource can be likened to a wrapper around a `PodSpec`, but users
+need to define a `PodSpec` for both Launcher and Worker containers.
+
+See the [`MPIJobSpec`
+definition](https://github.com/kubeflow/mpi-operator/blob/v0.4.0/pkg/apis/kubeflow/v2beta1/types.go#L137)
+for more details on what can be configured for an MPI application.
+
+It's important to bear in mind that the NNF Software is designed to override specific values within
+the `MPIJobSpec` for ensuring the desired behavior in line with NNF software requirements. To
+prevent complications, it's advisable not to delve too deeply into the specification. A few
+illustrative examples of fields that have been preconfigured include:
```

Comment:
Clarify whether those are preconfigured by mpi-operator, or by the NNF software.